### PR TITLE
feat(table): new table component

### PR DESCRIPTION
**Pull request by yqrashawn**
---

## Checklist

- [x] Fix linting errors
- [x] Tests have been added / updated (or snapshots)

## Change information

use ant-design's rc table

### CODE REVIEW

**Comment by AngeliaQ** [comment]
Task linked: [CU-3tqu41 table components](https://app.clickup.com/t/3tqu41)

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -8,3 +8,5 @@ components/select/* @angelia-yuqi-personal
 components/button/* @chain-txz
 components/card/* @angelia-yuqi-personal
 components/tooltip/* @yqrashawn
+components/table/* @yqrashawn
+pages/components/*/table.mdx @yqrashawn
```
Comment (AngeliaQ) [inline .github/CODEOWNERS]: you can add others component mdx codeowner together

File: components/table/table.tsx [hunk 2]
```diff
@@ -1,109 +1,384 @@
...
 
-type TableComponent<P = {}> = React.FC<P> & {
-  Column: typeof TableColumn
-}
-type ComponentProps = Partial<typeof defaultProps> &
-  Omit<Props, keyof typeof defaultProps> &
-  NativeAttrs
-;(Table as TableComponent<ComponentProps>).defaultProps = defaultProps
+Table.defaultProps = defaultProps
+Table.Column = RcTable.Column
```
Comment (AngeliaQ) [inline components/table/table.tsx:377]: may be written in index.ts

File: components/table/index.ts [hunk 3]
```diff
@@ -1,6 +1,3 @@
 import Table from './table'
-import TableColumn from './table-column'
-
-Table.Column = TableColumn
```
Comment (AngeliaQ) [inline components/table/index.ts:4]: why remove this and move to table.ts

**Review by AngeliaQ — DISMISSED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (yqrashawn) [reply .github/CODEOWNERS]: Already added them in the popover pr

File: components/table/index.ts (hunk 3 above)
Comment (yqrashawn) [reply components/table/index.ts:4]: Less code, no need to write import RcTable twice

**Review by AngeliaQ — DISMISSED** [review]

**Review by AngeliaQ — DISMISSED** [review]

**Review by AngeliaQ — DISMISSED** [review]

**Review by GuichiZhao — DISMISSED** [review]

**Review by zzlune — APPROVED** [review]

**Review by AngeliaQ — APPROVED** [review]

**Pull request merged**